src/ingester/parser/indexer_events.rs

```diff
@@ -3,25 +3,127 @@
 use anchor_lang::prelude::*;
 use light_compressed_account::indexer_event::event::{BatchNullifyContext, NewAddress};
 
-#[derive(Debug, PartialEq, Default, Clone, AnchorSerialize, AnchorDeserialize)]
+#[derive(Debug, PartialEq, Eq, Default, Clone, AnchorSerialize, AnchorDeserialize)]
 pub struct OutputCompressedAccountWithPackedContext {
     pub compressed_account: CompressedAccount,
     pub merkle_tree_index: u8,
 }
 
 #[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, Default, Eq, PartialEq)]
-pub struct MerkleTreeSequenceNumber {
+pub struct MerkleTreeSequenceNumberV2 {
+    pub tree_pubkey: Pubkey,
+    pub queue_pubkey: Pubkey,
+    pub tree_type: u64,
+    pub seq: u64,
+}
+
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, Default, Eq, PartialEq)]
+pub struct MerkleTreeSequenceNumberV1 {
     pub pubkey: Pubkey,
     pub seq: u64,
 }
 
-#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, Default, PartialEq)]
-pub struct PublicTransactionEvent {
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, Eq, PartialEq)]
+pub enum MerkleTreeSequenceNumber {
+    V1(MerkleTreeSequenceNumberV1),
+    V2(MerkleTreeSequenceNumberV2),
+}
+
+impl MerkleTreeSequenceNumber {
+    pub fn tree_pubkey(&self) -> Pubkey {
+        match self {
+            MerkleTreeSequenceNumber::V1(x) => x.pubkey,
+            MerkleTreeSequenceNumber::V2(x) => x.tree_pubkey,
+        }
+    }
+    pub fn seq(&self) -> u64 {
+        match self {
+            MerkleTreeSequenceNumber::V1(x) => x.seq,
+            MerkleTreeSequenceNumber::V2(x) => x.seq,
+        }
+    }
+}
+
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize, Eq, PartialEq)]
+pub enum PublicTransactionEvent {
+    V1(PublicTransactionEventV1),
+    V2(PublicTransactionEventV2),
+}
+
+impl PublicTransactionEvent {
+    pub fn input_compressed_account_hashes(&self) -> Vec<[u8; 32]> {
+        match self {
+            PublicTransactionEvent::V1(x) => x.input_compressed_account_hashes.clone(),
+            PublicTransactionEvent::V2(x) => x.input_compressed_account_hashes.clone(),
+        }
```

Comment:
could we return references here instead of cloning?
```suggestion
    pub fn input_compressed_account_hashes(&self) -> &Vec<[u8; 32]> {
        match self {
            PublicTransactionEvent::V1(x) => x.input_compressed_account_hashes.clone(),
            PublicTransactionEvent::V2(x) => x.input_compressed_account_hashes.clone(),
        }
```

Reply:
If it works it pronbably makes sense for all getters except for sequence numbers.